domains/client/domain-operator/src/fetch_domain_bootstrap_info.rs

```diff
@@ -39,13 +46,48 @@ where
         .runtime_api()
         .domain_instance_data(best_hash, self_domain_id)?
     {
+        let domain_best_number = consensus_client
+            .runtime_api()
+            .domain_best_number(best_hash, self_domain_id)?
+            .unwrap_or_default();
+
+        // The `domain_best_number` is the expected best domain block after the operator has
+        // processed the consensus block at `best_hash`. If `domain_best_number` is not zero
+        // and the domain chain is not started, the domain block `0..domain_best_number` are
+        // missed, we can not preceed running as a domain node because:
+        //
+        // - The consensus block and state that derive the domain block `0..domain_best_number`
+        //    may not available anymore
+        //
+        // - There may be domain runtime upgrade in `0..domain_best_number` which will result
+        //    in inconsistent `raw_genesis`
+        if !is_domain_started && !domain_best_number.is_zero() {
+            return Err(
+                "Unable to restart the existing consensus node as a domain node"
```

Comment:
Do we want to recommend wiping the db and domain directories here?
```suggestion
                "An existing consensus node can’t be restarted as a domain node"
```

Reply:
Ideally yes